modules: Kconfig: Provide default definitions for modules with blobs

jhedberg opened the pull request:
There are HCI drivers which reference the module-specific Kconfig symbols which are auto-generated when the modules are present. Provide default definitions for these symbols in case the modules are missing, so that compliance checking doesn't stumble over seemingly non-existent Kconfig options.

Fixes #78056

jhedberg committed:
modules: Kconfig: Provide default definitions for modules with blobs

There are HCI drivers which reference the module-specific Kconfig symbols
which are auto-generated when the modules are present. Provide default
definitions for these symbols in case the modules are missing, so that
compliance checking doesn't stumble over seemingly non-existent Kconfig
options.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

erwango reviewed (commented):
Fine to me but can this be documented somehow ?

jhedberg committed:
doc: modules: Document conventions for auto-generated Kconfig options

Document the recommended practice regarding having default definitions for
module-specific Kconfig symbols.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

jhedberg commented:
> Fine to me but can this be documented somehow ?

@erwango I added a second commit for this.

kartben commented on doc/develop/modules.rst:
```diff
@@ -621,6 +621,14 @@ Zephyr modules
 
+Additionally, ``ZEPHYR_<MODULE_NAME>_MODULE`` and ``ZEPHYR_<MODULE_NAME>_MODULE_BLOBS``
+(in case the module declares blobs) symbols are automatically generated for available
+modules. These can be used e.g. do declare dependencies from other Kconfig symbols
```
```suggestion
modules. These can be used e.g. to declare dependencies from other Kconfig symbols
```

sylvioalves reviewed (approved):
LGTM

tejlmand reviewed (approved):
lgtm.

Now that we always have to define the symbol `ZEPHYR_<name>_MODULE` and `ZEPHYR_<name>_MODULE_BLOBS` then I wonder if we should revert this commit ?
https://github.com/zephyrproject-rtos/zephyr/pull/77476/commits/90f508defe0ca8ec4e4fb9b0db54038db8f43dfa

jhedberg commented:
> Now that we always have to define the symbol `ZEPHYR_<name>_MODULE` and `ZEPHYR_<name>_MODULE_BLOBS` then I wonder if we should revert this commit ?
> [90f508d](https://github.com/zephyrproject-rtos/zephyr/commit/90f508defe0ca8ec4e4fb9b0db54038db8f43dfa)

Well, I used the word "recommended" in the documentation that I added, so not sure. Also, it's a bit unclear to me where exactly that commit helped, considering that we had the need for this PR to make compliance pass (on downstream trees which may omit certain modules).
